### PR TITLE
backport: custom params serializer support for v0.x

### LINKED ISSUE #6262: paramsSerializer is not working in v0.28.0
### Describe the bug

Although there's a backport in #6227, in v0.28.0, the serializer isn't working as expected. This is because the v0.28.0 version of `buildUrl` ([code](https://github.com/axios/axios/blob/3b7635aefc842c05da0ec8c90e8bd09cb54616b8/lib/helpers/buildURL.js#L36-L40)) is simply only taking `options.encode`, not taking `options.serializer` into account:

https://github.com/axios/axios/blob/ce46346c0f5306b65023bba324c76fe5fe83eee8/lib/helpers/buildURL.js#L36-L44

This seems to be a refactor to match the v1.x version, because in v0.27.2 at least ([code](https://github.com/axios/axios/blob/v0.27.2/lib/helpers/buildURL.js#L28-L30)), it takes the `paramSerializer` function into account:

https://github.com/axios/axios/blob/bc733fec78326609e751187c9d453cee9bf1993a/lib/helpers/buildURL.js#L28-L30

In order for #6227 to actually work, there needs to be a backport similar to #5113 towards v0.x

### To Reproduce

- Set the version to v0.28.0
- Set the `paramsSerializer` function
- It won't work

### Code snippet

```js
// Simply following https://axios-http.com/docs/req_config

axios.create({
  paramsSerializer: function (params) {
    return Qs.stringify(params, {arrayFormat: 'brackets'})
  },
})
```

### Expected behavior

The parameters should be serialized, but it didn't

### Axios Version

v0.28.0

### Adapter Version

_No response_

### Browser

_No response_

### Browser Version

_No response_

### Node.js Version

_No response_

### OS

_No response_

### Additional Library Versions

_No response_

### Additional context/Screenshots

_No response_

### PR DESCRIPTION
**Pull request by briwa**
Resolves #6262

This will add similar support to #5113 (`paramsSerializer: { serializer? }` or `paramsSerializer(params) { .. }`) which was missing in v0.x.

Was hoping if this could be released as a patch for v0.28.x

### CODE REVIEW

**Review by JadeMugs — CHANGES_REQUESTED** [review]

**Comment by JadeMugs** [comment]
Hey @briwa, watching your MR I see the implementation is slightly different from the one in v1.x. IMMO this choice could create more friction for users who try to upgrade Axios from v0.x to v1.x.

File: lib/helpers/buildURL.js
Comment (briwa) [reply lib/helpers/buildURL.js:50]: @JadeMugs I disagree. I don't think it matters much since the `hashMark` part of the code is only to strip out the URL. Besides, I'm following the version in v1.x here: https://github.com/DigitalBrainJS/axios/blob/92f0758429541b05c614dc25bc712379d8a41dbd/lib/helpers/buildURL.js

File: test/typescript/axios.ts
Comment (briwa) [reply test/typescript/axios.ts:26]: @JadeMugs Thanks, I missed it. I just added them

**Comment by briwa** [comment]
> Hey @briwa, watching your MR I see the implementation is slightly different from the one in v1.x. IMMO this choice could create more friction for users who try to upgrade Axios from v0.x to v1.x.

@JadeMugs Hey, I disagree that it will cause friction because as I mentioned in the PR description, this PR is simply following the same patch that was introduced in https://github.com/axios/axios/pull/5113. If this patch would cause friction, then the original patch would be, too, since I'm not doing anything different. But I might be wrong, so I'll let the maintainers decide.

File: lib/helpers/buildURL.js
Comment (briwa) [reply lib/helpers/buildURL.js:50]: @JadeMugs I disagree. I don't think it matters much since the `hashMark` part of the code is only to strip out the URL. Besides, I'm following the version in v1.x here: 

https://github.com/DigitalBrainJS/axios/blob/92f0758429541b05c614dc25bc712379d8a41dbd/lib/helpers/buildURL.js#L39-L57

Comment (briwa) [reply lib/helpers/buildURL.js:50]: @JadeMugs I disagree. I don't think it matters much since the `hashMark` part of the code is only to strip out the URL. Besides, I'm following the version in https://github.com/axios/axios/pull/5113 here: 

https://github.com/DigitalBrainJS/axios/blob/92f0758429541b05c614dc25bc712379d8a41dbd/lib/helpers/buildURL.js#L39-L57

**Pull request merged**
